app/scripts/modules/core/src/search/global/GlobalSearch.tsx

```diff
@@ -0,0 +1,405 @@
+                return null;
+              }
+
+              this.resultRefs = range(recentCategories.length).map(() => []);
+
+              return (
+                <ul className="dropdown-menu" role="menu">
+                  {recentCategories.map((category, categoryIndex) => ([
+                    <li key={category.category} className="category-heading">
+                      <div className="category-heading">Recent {category.category}</div>
+                    </li>,
+                    ...category.results.map((result, index) => {
+                      const params = result.params || {};
+                      const account = result.account || params.account || params.accountId || params.accountName;
+
+                      return (
+                        <li
+                          key={result.id}
+                          className="result"
+                          onClick={() => {
+                            this.hideResults();
+                            ReactGA.event({ category: 'Global Search', action: 'Recent Item Selected' });
+                          }}
+                        >
+                          <UISref to={result.state} params={result.params}>
+                            <a
+                              ref={(ref) => {
+                                if (this.resultRefs[categoryIndex]) {
+                                  this.resultRefs[categoryIndex][index] = ref;
+                                }
+                              }}
+                              onKeyDown={this.navigateResult}
+                            >
+                              <SearchResult displayName={result.displayName} account={account} />
+                            </a>
+                          </UISref>
+                        </li>
+                      );
+                    })
+                  ]))}
+                </ul>
+              );
+            }}
+          />
+        }
+        {!!query && !querying && showSearchResults &&
+          <ul className="dropdown-menu" role="menu">
+            {categories.map((category, categoryIndex) => ([
+              <li key={category.id} className="category-heading">
+                <div className="category-heading">{category.category}</div>
+              </li>,
+              category.results.map((result, index) => (
+                <li className="result">
+                  <a
+                    onKeyDown={this.navigateResult}
+                    onClick={() => {
+                      ReactGA.event({ category: 'Global Search', action: 'Result Selected' });
+                      this.hideResults();
+                      this.clearFilters(result);
+                    }}
+                    ref={(ref) => {
+                      if (this.resultRefs[categoryIndex]) {
+                        this.resultRefs[categoryIndex][index] = ref;
+                      }
+                    }}
+                    href={result.href}
```

Comment:
it seems like both these and the `see more results` link should get moved to a `UISRef` in a subsequent change.